src/legacy/server/saved_objects/service/lib/search_dsl/search_dsl.test.ts

```diff
@@ -20,47 +20,46 @@
 jest.mock('./query_params');
 jest.mock('./sorting_params');
 
-import { getSearchDsl } from './search_dsl';
 import * as queryParamsNS from './query_params';
+import { getSearchDsl } from './search_dsl';
 import * as sortParamsNS from './sorting_params';
 
+const getQueryParams = queryParamsNS.getQueryParams as jest.Mock;
+const getSortingParams = sortParamsNS.getSortingParams as jest.Mock;
+
+const MAPPINGS = { properties: {} };
+const SCHEMA = {
+  isHiddenType: (type: string) => false,
+  isNamespaceAgnostic: (type: string) => type === 'global',
+};
+
 describe('getSearchDsl', () => {
   afterEach(() => {
-    queryParamsNS.getQueryParams.mockReset();
-    sortParamsNS.getSortingParams.mockReset();
+    getQueryParams.mockReset();
+    getSortingParams.mockReset();
   });
 
   describe('validation', () => {
     it('throws when type is not specified', () => {
       expect(() => {
-        getSearchDsl(
-          {},
-          {},
-          {
-            type: undefined,
-            sortField: 'title',
-          }
-        );
+        getSearchDsl(MAPPINGS, SCHEMA, {
+          type: undefined as any,
```

Comment:
Not sure how to test null types that could be passed in from non-typescript files?

Reply:
sorry, I didn't get the question

Reply:
This is an interesting scenario where within the function, it throws an error if `type` is not defined.

```
if (!type) {
  throw Boom.notAcceptable('type must be specified');
}
```
So it would make sense to make `type` mandatory in TypeScript. When it comes to tests, this test ensures the error is thrown when passing `undefined` but now in TypeScript the only way I can make this test still work is by doing `undefined as any`. The question is if that's an ok path to go with? There's still some places that use the code outside of TypeScript that may still need the error to be thrown.

Reply:
don't see a problem. But there is only one place calling `getSearchDsl`, as an alternative solution we can lift this check to the `*.js` file.
at least defining `type` as a required field looks right.